### ISSUE TITLE
Keras Issue concatenate single input instead of multiple

### ISSUE THREAD

**Issue by aezco** [keras-team/keras#12871]
I am trying similar things, but I am getting stuck with the input of the model.

 `  

     def generate_generator_multiple(generator,dir1, dir2,dir3, batch_size, img_size):
          genX1 = generator.flow_from_directory(dir1,
                                          target_size = (img_size,img_size),
                                          class_mode = 'categorical',
                                          batch_size = batch_size,
                                          shuffle=False, 
                                          seed=7,subset='training')

          genX2 = generator.flow_from_directory(dir2,
                                          target_size = (img_size,img_size),
                                          class_mode = 'categorical',
                                          batch_size = batch_size,
                                          shuffle=False, 
                                          seed=7,subset='training')

          genX3 = generator.flow_from_directory(dir3,
                                      target_size = (img_size,img_size),
                                      class_mode = 'categorical',
                                      batch_size = batch_size,
                                      shuffle=False, 
                                      seed=7,subset='training')
           while True:
                  X1i = genX1.next()
                  X2i = genX2.next()
                  X3i = genX3.next()
           yield [X1i[0], X2i[0],X3i[0]], X3i[1]  #Yield both images and their mutual label

    train_datagen = ImageDataGenerator(validation_split=VAL_SPLIT,rescale=1./255)
    inputgenerator=generate_generator_multiple(generator=train_datagen,
                                           dir1='mergedata',
                                           dir2='mergedata',
                                           dir3='mergedata',
                                           batch_size=BATCH_SIZE,
                                           img_size=IMG_SIZE)   

    modelA = load_model('/content/8-base_model_1.h5')
    modelA.load_weights('/content/8-weights.002-0.991-0.406.hdf5') 
    
    modelB = load_model('/content/30-base_model_1.h5')
    modelB.load_weights('/content/30-weights.010-0.499-0.374.hdf5')
    
    modelC = load_model('/content/80-base_model_1.h5')
    modelC.load_weights('/content/80-weights.002-1.064-0.305.hdf5')
    for layer in modelA.layers:
        layer.trainable = False
        layer.name = layer.name + str("_1")
      
    for layer in modelB.layers:
        layer.trainable = False
        layer.name = layer.name + str("_2")
        
    for layer in modelC.layers:
        layer.trainable = False
        layer.name = layer.name + str("_3")

    mod_1 = concatenate([modelA.layers[-1].output,  modelB.layers[-1].output,modelC.layers[-1].output])     
    mod_1 = Dropout(0.2)(mod_1)
    mod_1 = Dense(15, activation='relu')(mod_1)
    predictions = Dense(9, activation='softmax', name='pred_age')(mod_1)
    
    top_model = Model(inputs=[modelA.input, modelB.input, modelC.input], outputs=predictions)

    opt = get_optimizer('adam', lr_rate)
    top_model.compile(loss=['categorical_crossentropy'],
              optimizer=opt,
              metrics=['accuracy','mae'])

    model.fit_generator(inputgenerator,
                               epochs=EPOCHS,
                               steps_per_epoch = STEPS_PER_EPOCH,
                               validation_data=validgenerator,
                               validation_steps = VALIDATION_STEPS,
                               verbose=1,
                               callbacks=callbacks)`

I first trained 3 different models:

- model 1 = 3 labels (8,9,10)
- model 2 = 2 labels (30,31)
- model 3 = 4 labels (80,81,82,83)

I would like to combine these 3 models into a final model with 1 output containing 9 labels (8,9,10,30,31,80,81,82,83). The final input needs to get 1 input image instead of 3 images. But I still get stuck. I am building a kind of hierarchy here to improve the accuracy.
